packages/block-library/src/navigation-link/link-ui.js

```diff
@@ -177,14 +180,82 @@ export function LinkUI( props ) {
 
 	// Memoize link value to avoid overriding the LinkControl's internal state.
 	// This is a temporary fix. See https://github.com/WordPress/gutenberg/issues/50976#issuecomment-1568226407.
-	const link = useMemo(
-		() => ( {
-			url,
-			opensInNewTab,
-			title: label && stripHTML( label ),
-		} ),
-		[ label, opensInNewTab, url ]
-	);
+	// const link = useMemo(
+	// 	() => ( {
+	// 		url,
+	// 		opensInNewTab,
+	// 		title: label && stripHTML( label ),
+	// 	} ),
+	// 	[ label, opensInNewTab, url ]
+	// );
+
+	// ...( newUrl && { url: encodeURI( safeDecodeURI( newUrl ) ) } ),
+	// ...( label && { label } ),
+	// ...( undefined !== opensInNewTab && { opensInNewTab } ),
+	// ...( id && Number.isInteger( id ) && { id } ),
+	// ...( kind && { kind } ),
+	// ...( type && type !== 'URL' && { type } ),
+
+	const { toLink, toData } = getLinkValueTransforms( {
+		url: {
+			dataKey: 'url',
+			toData: ( value ) => encodeURI( safeDecodeURI( value ) ),
+		},
+		title: {
+			dataKey: 'label',
+			toLink: ( value ) => stripHTML( value ),
+			toData: (
+				newLabel = '',
+				{ url: newUrl = '' },
+				{ label: originalLabel = '' }
```

Comment:
Whilst terse, this does require the reader to understand the API of the toData reducer.

Instead let's define these as well named arguments (e.g. `fullLinkValue`, `currentDataValue`) and _then_ destructure from them.